pkgs/applications/editors/vscode/extensions/hashicorp.terraform/default.nix

```diff
@@ -2,20 +2,29 @@
   lib,
   vscode-utils,
   terraform-ls,
+  vscode-extension-update-script,
 }:
+
 vscode-utils.buildVscodeMarketplaceExtension {
   mktplcRef = {
     name = "terraform";
     publisher = "hashicorp";
-    version = "2.34.3";
-    hash = "sha256-kE9xH0cp741aCqhrwFDW+lQxOAsdLNzCCOTWMxd+li0=";
+    version = "2.34.4";
+    hash = "sha256-gTU37q7kh51CxqQE7KylSwEK4ubiUU5uihbFew/Fg98=";
   };
 
   postPatch = ''
     substituteInPlace dist/extension.js \
       --replace-fail 'this.customBinPath=Ga("terraform").get("languageServer.path")' 'this.customBinPath = Ga("terraform").get("languageServer.path") || '${terraform-ls}/bin/terraform-ls';'
```

Comment:
Can't we just override the binary location with a config setting ?

Reply:
I believe adding:

```nix
  postInstall = ''
    cd "$out/$installPrefix"
    jq '.contributes.configuration[0].properties."terraform.languageServer.path".default = "${terraform-ls}/bin/terraform-ls"' package.json | sponge package.json
  '';
```

Is simply enough.

Reply:
> Thanks for the further explanation. Would love to hear from the maintainers. @rhoriguchi

The reason I've patched the extension, is that it just works like on any other distro. But overriding the config default in the extension would have the same effect.